cmd/view-fits/main.go

```diff
@@ -0,0 +1,205 @@
+Navigation:
+right/left arrows: switch to previous/next file
+up/down arrows: switch to previous/next image in the current file
+`)
+	}
+
+	flag.Parse()
+
+	if *help || len(os.Args) < 2 {
+		flag.Usage()
+		os.Exit(0)
+	}
+
+	// Parsing input files.
+	for _, fname := range flag.Args() {
+
+		finfo := fileInfo{Name: fname}
+
+		// Opening the file.
+		if r, err = os.Open(fname); err != nil {
+			log.Fatal("Can not open the input file: %s.", err)
+		}
+
+		// Opening the FITS file.
+		if f, err = fitsio.Open(r); err != nil {
+			log.Fatal("Can not open the FITS input file: %s.", err)
+		}
+		defer f.Close()
+
+		// Getting the file HDUs.
+		hdus := f.HDUs()
+		for _, hdu := range hdus {
+			// Getting the header informations.
+			header := hdu.Header()
+			axes := header.Axes()
+
+			// Leaving HDU with no axes.
+			if len(axes) != 0 {
+				// Keeping HDU with IMAGE_HDU type.
+				if HDUType := hdu.Type(); HDUType == fitsio.IMAGE_HDU {
+					// Filling the images array.
+					finfo.Images = append(finfo.Images, hdu.(fitsio.Image).Image())
+				}
+			}
+		}
+
+		if len(finfo.Images) > 0 {
+			infos = append(infos, finfo)
+		}
+	}
+
+	if len(infos) == 0 {
+		log.Fatal("The input file contains no FITS images.")
+	}
+
+	driver.Main(func(s screen.Screen) {
+
+		// Number of files.
+		nbFiles := len(infos)
+
+		// Current displayed file and image in file.
+		ifile := 0
+		iimg := 0
+
+		// Building the main window.
+		w, err := s.NewWindow(&screen.NewWindowOptions{})
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer w.Release()
+		// Building the screen buffer.
+		b, err := s.NewBuffer(image.Point{})
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer b.Release()
+
+		w.Fill(b.Bounds(), color.Black, draw.Src)
+		w.Publish()
+
+		for {
+			e := w.NextEvent()
+
+			switch e := e.(type) {
+			default:
+			case key.Event:
+				switch e.Code {
+				case key.CodeEscape, key.CodeQ:
+					return
+				case key.CodeRightArrow:
+					if e.Direction == key.DirPress {
+						if ifile < nbFiles-1 {
+							ifile++
+						} else {
+							ifile = 0
+						}
+						iimg = 0
+						b.Release()
+						w.Send(paint.Event{})
+					}
+				case key.CodeLeftArrow:
+					if e.Direction == key.DirPress {
+						if ifile == 0 {
+							ifile = nbFiles - 1
+						} else {
+							ifile--
+						}
+						iimg = 0
+						b.Release()
+						w.Send(paint.Event{})
+					}
+
+				case key.CodeUpArrow:
+					if e.Direction == key.DirPress {
+						nbImg := len(infos[ifile].Images)
+						if iimg < nbImg-1 {
+							iimg++
+						} else {
+							iimg = 0
+						}
+						b.Release()
+						w.Send(paint.Event{})
+
+					}
+				case key.CodeDownArrow:
+					if e.Direction == key.DirPress {
+						nbImg := len(infos[ifile].Images)
+						if iimg == 0 {
+							iimg = nbImg - 1
+						} else {
+							iimg--
+						}
+						b.Release()
+						w.Send(paint.Event{})
+
+					}
+				}
+
+			case paint.Event:
+				img := infos[ifile].Images[iimg]
+
+				b, err = s.NewBuffer(img.Bounds().Size())
```

Comment:
If I am not mistaken, I believe you should also `defer b.Release()` after the `err` check.